prep/wiki/wiki

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+# The table of contents HTML looks like this:
+# N = level (main, subsection, etc.  level 1 contains the main sections, 2 contains subsections
+# K = ordinal in table of contents (toc)
+# ul#mw-panel-toc-list
+#   li#toc-* .vector-toc-list-item .vector-toc-level-N
+#     a.vector-toc-link
+#     div .vector-toc-text
+#       span .vector-toc-numb
+#         K
+#       TOPIC
+#
+# slugify input to make curl/xmllint easier
+title=$(echo ${1:-} | tr " " _ | tr "[:upper:]" "[:lower:]" | awk '{print toupper(substr($0,1,1))substr($0,2)}')
+section=$(echo ${2:-} | tr " " _ | tr "[:upper:]" "[:lower:]" | awk '{print toupper(substr($0,1,1))substr($0,2)}')
+
+# saving the page and re-using it is a lot easier than trying to massage
+# xmllint to give me nice output
+page=$(curl -sL "https://en.wikipedia.org/wiki/$title")
+if [[ -z "$section" ]]; then
+	# first sentence
+	xmllint --htmlout --html \
+		--xpath '//div[@id="mw-content-text"]/div[@class="mw-parser-output"]/p[not(contains(@class, "mw-empty-elt"))][1]/descendant-or-self::*/text()' \
+		- <<<"$page" 2>/dev/null |
```

Comment:
Making use of herestrings to pass the variable in as a file

https://tldp.org/LDP/abs/html/x17837.html

I had this as a process substitution beforehand but once I needed the page twice then I moved to this.